### PR TITLE
[10.x] Adding Pipeline Facade

### PR DESCRIPTION
**Pull request by byt3sage**
Hey All!
I have a feeling there is a distinct reasoning behind why this hasn't been done before but here it goes. I love pipelines... I also love facades... Why not merge the two and have a pipeline facade?

This is the simplest implementation I could think of. It almost feels too simple so I'm sure there's some form of logic I have missed.

I'd love some feedback to try get this pushed into Laravel as a core facade!

Happy for this to be closed if it's a non-starter!

P.S - I'll be doing some light tweaking of bits, just thought I'd push this up first so I don't waste my time if it's never going to be accepted :)

### CODE REVIEW

File: src/Illuminate/Pipeline/PipelineServiceProvider.php [hunk 1]
```diff
@@ -16,7 +16,15 @@ class PipelineServiceProvider extends ServiceProvider implements DeferrableProvi
     public function register()
     {
         $this->app->singleton(
-            PipelineHubContract::class, Hub::class
+            PipelineHubContract::class,
+            Hub::class
+        );
+
+        $this->app->singleton(
```
Comment (rodrigopedra) [inline src/Illuminate/Pipeline/PipelineServiceProvider.php:23]: Why is this a singleton? Wouldn't it keep pipes from previous usage? `->through()` replace the pipes, but `->pipe()` doesn't

Comment (jaetoole) [reply src/Illuminate/Pipeline/PipelineServiceProvider.php:23]: Very true point!

Comment (rojtjo) [reply src/Illuminate/Pipeline/PipelineServiceProvider.php]: Currently the Facade still kind of acts as a singleton due to the resolved instances being cached on a static property as seen here:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Support/Facades/Facade.php#L228-L241

To work around this you would also need to change the static `$cached` property to `false`:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Support/Facades/Facade.php#L31-L36

**Comment by JordanDalton** [comment]
I've proposed a helper (https://github.com/laravel/framework/pull/46198) but haven't heard back from the team that approves PRs.

**Comment by jaetoole** [comment]
Ah! Would be good to link these PRs then!I’ll update the caching of the facade later this evening!On 25 Feb 2023, at 17:32, Jordan Dalton ***@***.***> wrote:﻿
I've proposed a helper (#46198) but haven't heard back from the team that approves PRs.

**Comment by jaetoole** [comment]
So I've disabled caching on the facade (I totally didn't even know what was a feature!) - happy to take suggestions from anyone else as to what needs doing to get this merged :)

**Review by rodrigopedra — COMMENTED** [review]
LGTM

**Comment by taylorotwell** [comment]
You said you will be doing some "light tweaking". I can't review the PR until it is totally complete. Please mark as ready for review when that is done.

**Comment by jaetoole** [comment]
Hey, sorry - it’s ready for review nowSent from my iPhoneOn 2 Mar 2023, at 19:10, Taylor Otwell ***@***.***> wrote:﻿
You said you will be doing some "light tweaking". I can't review the PR until it is totally complete. Please mark as ready for review when that is done.

**Comment by jaetoole** [comment]
Just to re-iterate, the light tweaking was just making sure the correct docblocks etc were added - hence I requested a re-review from the people who raised additional comments :)

**Pull request merged**